hubspot_sync/tasks_test.py

```diff
@@ -426,7 +426,9 @@ def test_sync_failed_contacts(mocker):
     )
     result = tasks.sync_failed_contacts(user_ids)
     assert mock_sync.call_count == 4
-    assert result == [user_ids[1], user_ids[3]]
+
+    # Expect two failed contacts that correspond with the ApiExceptions above.
+    assert len(result) == 2
```

Comment:
Would this be better if it checked for the IDs in the result (`user_ids[1] in result`)?

Reply:
I think the issue that causes this test to be flaky is the `filter()` in `sync_failed_contacts()` which doesn't guarantee the order of the users.  So `filter()` might return [user_ids[1], user_ids[2], user_ids[3], user_ids[4] which would mean that user_ids[1] and user_ids[3] are going to experience an ApiException, `filter()` could also return [user_ids[2], user_ids[4], user_ids[3], user_ids[1] which would mean that user_ids[2] and user_ids[3] are going to experience an ApiException and be returned by `sync_failed_contacts()`.